code/game/objects/items/weapons/storage/lockbox.dm

```diff
@@ -62,6 +62,34 @@
 		for(var/mob/O in viewers(user, 3))
 			O.show_message(text("<span class='notice'>\The [src] has been broken by [] with an electromagnetic card!</span>", user), 1, text("You hear a faint electrical spark."), 2)
 			return
+
+/obj/item/weapon/storage/lockbox/emp_act(severity)

```

Comment:
When I was talking about the EMP's I was more meaning a overall change with all EMP's.
The use of C.corrupt() there is no way to actually fix the weapon if it gets corrupted. This makes it heavily RNG that science will unlock a broken weapon. 
The same issue actually already occurs with miners and their mining weapon when a explosion occurs near enough their ranged weapon breaks due to it not having enough power to even shoot once. Your idea is nice and will mean for powergamers this will be actually discouraging for them.